src/components/legend/defaults.js

```diff
@@ -4,20 +4,39 @@ var Registry = require('../../registry');
 var Lib = require('../../lib');
 var Template = require('../../plot_api/plot_template');
 
+var plotsAttrs = require('../../plots/attributes');
 var attributes = require('./attributes');
 var basePlotLayoutAttributes = require('../../plots/layout_attributes');
 var helpers = require('./helpers');
 
 
 module.exports = function legendDefaults(layoutIn, layoutOut, fullData) {
     var containerIn = layoutIn.legend || {};
+    var containerOut = Template.newContainer(layoutOut, 'legend');
```

Comment:
Is it possible to keep this `Template.newContainer` call where it was below the [short-circuit return calls](https://github.com/plotly/plotly.js/pull/6040/files#diff-1c46ac7461e74f0614d86dca7388a3e60d2cf4164a9691913425da157f03eafeR86) so that we don't create the `fullLayout.legend` container when there won't be a legend at all? Or in some other way only create it once we know it'll be used?

Reply:
Good call. I was not able keep the `newContainer` call where it was; but instead deleted the `legend` in cases where there is no legend to show. See https://github.com/plotly/plotly.js/pull/6040/commits/f3095bf04f6f08c1df478c47273ae7ec64a24956.